L1Trigger/L1THGCal/src/be_algorithms/HGCalMulticlusteringImpl.cc

```diff
@@ -219,28 +208,38 @@ void HGCalMulticlusteringImpl::clusterizeDBSCAN( const edm::PtrVector<l1t::HGCal
     }
     /* making the collection of multiclusters */
     for( unsigned i(0); i<multiclustersTmp.size(); ++i ){
-        math::PtEtaPhiMLorentzVector calibP4( multiclustersTmp.at(i).pt() * calibSF_, 
-                multiclustersTmp.at(i).eta(), 
-                multiclustersTmp.at(i).phi(), 
+
+        // compute the eta, phi observables for multicluster starting from its barycenter x,y,z position + pT as scalar sum of pT of constituents
+        double sumPt=0.;
+
+        const std::vector<edm::Ptr<l1t::HGCalCluster>> &pertinentClu = multiclustersTmp.at(i).constituents();
+        for( std::vector<edm::Ptr<l1t::HGCalCluster>>::const_iterator  it_clu=pertinentClu.begin(); it_clu<pertinentClu.end(); it_clu++) sumPt +=(*it_clu)->pt();
+
+        math::PtEtaPhiMLorentzVector multiclusterP4(  sumPt,
+                multiclustersTmp.at(i).centre().eta(),
+                multiclustersTmp.at(i).centre().phi(),
                 0. );
-        // overwriting the 4p with the calibrated 4p     
-        multiclustersTmp.at(i).setP4( calibP4 );
+        multiclustersTmp.at(i).setP4( multiclusterP4 );
+
 
         if( multiclustersTmp.at(i).pt() > ptC3dThreshold_ ){
 
             //compute shower shape
             multiclustersTmp.at(i).showerLength(shape_.showerLength(multiclustersTmp.at(i)));
+            multiclustersTmp.at(i).coreShowerLength(shape_.coreShowerLength(multiclustersTmp.at(i), triggerGeometry));
```

Comment:
using `at(i)` every time is unnecessarily slow - just check the size once and then use `[i]`

Reply:
done